code/model/SiteTree.php

```diff
@@ -1078,39 +1078,13 @@ public function canEdit($member = null) {
 	}
 
 	/**
-	 * This function should return true if the current user can publish this page. It can be overloaded to customise
-	 * the security model for an application.
-	 *
-	 * Denies permission if any of the following conditions is true:
-	 * - canPublish() on any extension returns false
-	 * - canEdit() returns false
-	 *
-	 * @uses SiteTreeExtension->canPublish()
-	 *
-	 * @param Member $member
-	 * @return bool True if the current user can publish this page.
+	 * @deprecated
 	 */
-	public function canPublish($member = null) {
-		if(!$member || !(is_a($member, 'Member')) || is_numeric($member)) $member = Member::currentUser();
-		
-		if($member && Permission::checkMember($member, "ADMIN")) return true;
-
-		// Standard mechanism for accepting permission changes from extensions
-		$extended = $this->extendedCan('canPublish', $member);
-		if($extended !== null) return $extended;
-
-		// Normal case - fail over to canEdit()
-		return $this->canEdit($member);
-	}
-	
 	public function canDeleteFromLive($member = null) {
-		// Standard mechanism for accepting permission changes from extensions
-		$extended = $this->extendedCan('canDeleteFromLive', $member);
```

Comment:
I think we only need to show the deprecation notice if it's actually called. In which case, it'll get shown regardless of whether or not extensions define them.

If we never invoke this method, we can't show the notice otherwise.